### PR TITLE
Require D3 for browserify support

### PR DESCRIPTION
**Pull request by m-h-t**

### CODE REVIEW

File: src/core.js [hunk 1]
```diff
@@ -11,6 +11,11 @@ nv.charts = {}; //stores all the ready to use charts
 nv.logs = {}; //stores some statistics and potential error messages
 nv.dom = {}; //DOM manipulation functions
 
+// Node/CommonJS - require D3
+if (typeof(module) !== 'undefined' && typeof(exports) !== 'undefined' && typeof(d3) == 'undefined') {
+    d3 = require('d3');
+}
+
```
Comment (liquidpele) [inline src/core.js]: I'm not sure this is right... it's not ensuring that nvd3 loads after d3 is loaded...

Comment (m-h-t) [inline src/core.js]: You're talking about the order in which it is loaded? Or whether it's loaded at all? In which case do you think there is going to be a problem?

Comment (mqklin) [inline src/core.js]: @m-h-t , the problem is that `nvd3` requires `d3` to be in global space. It's hurt, because global namespacing is a bad pattern.

**Comment by mqklin** [comment]
Can anyone review this PR?

**Comment by mqklin** [comment]
Can anyone review this PR?

**Comment by mqklin** [comment]
I think it should be support not only for UMD, but also for AMD. I will try to make it.

**Comment by readcodelearn** [comment]
I can confirm that this PR works for browserify

**Comment by topicus** [comment]
I can confirm it works for browserfy.

**Comment by liquidpele** [comment]
Okay, well if it works for ya'll then I'll merge it in, thanks for confirming it :)

**Pull request merged**